### PR TITLE
[Whisper] add a fusion option to split input bias from MHA/DMHA

### PR DESCRIPTION
**Pull request by PeixuanZuo**
Whsiper model contains five different types of attention, q, k, v bias was fused into Attention/MHA/DMHA op,

encoderdecoderinit subgraph
- Attention: encoder attention
- Attention: decoder self attention + present k, v
- MultiHeadAttention: decoder cross attention + present k and v. q and v have bias.

decoder subgraph
- DecoderMultiHeadAttention: decoder cross attention + past k, v. q has bias
- DecoderMultiHeadAttention: decoder self attention + past/present k, v. q, k, v have bias.

For ROCm EP, MHA/DMHA doesn't support additional bias. This PR add a fusion option `disable_multi_head_attention_bias` to split q.k,v bias from MHA/DMHA.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/models/whisper/convert_to_onnx.py [hunk 1]
```diff
@@ -21,6 +21,11 @@
 
 logger = logging.getLogger("")
 
+PROVIDERS = {
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/convert_to_onnx.py:24]:
```suggestion
PROVIDERS = {
    "cpu": "CPUExecutionProvider",
    "cuda": "CUDAExecutionProvider",
    "rocm": "ROCMExecutionProvider",
}
```

File: onnxruntime/python/tools/transformers/models/whisper/convert_to_onnx.py [hunk 2]
```diff
@@ -185,6 +190,16 @@ def parse_arguments(argv=None):
         help="filepath to load pre-trained model with custom state dictionary (e.g. pytorch_model.bin)",
     )
 
+    parser.add_argument(
+        "-r",
+        "--provider",
+        required=False,
+        type=str,
+        default="cuda",
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/convert_to_onnx.py:198]: I think the default should be CPU for machines that don't have CUDA or ROCm

File: onnxruntime/python/tools/transformers/models/whisper/convert_to_onnx.py [hunk 3]
```diff
@@ -208,6 +223,7 @@ def export_onnx_models(
     quantize_per_channel: bool = False,
     quantize_reduce_range: bool = False,
     state_dict_path: str = "",
+    provider: str = "cuda",
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/convert_to_onnx.py:226]:
```suggestion
    provider: str = "cpu",
```

File: onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py [hunk 4]
```diff
@@ -230,13 +230,15 @@ def optimize_onnx(
         use_external_data_format: bool = False,
         auto_mixed_precision: bool = True,
         use_gpu: bool = False,
+        provider: str = "cuda",
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py:233]:
```suggestion
        provider: str = "cpu",
```

**Review by kunal-vaishnavi — APPROVED** [review]

**Review by cloudhan — APPROVED** [review]

**Pull request merged**